pkg/cloud/gcp/actuators/util/gcp_machine_architecture.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import "strings"
+
+type NormalizedArch string
+
+const (
+	// ArchitectureAmd64 is the normalized architecture name for amd64.
+	ArchitectureAmd64 NormalizedArch = "amd64"
+	// ArchitectureArm64 is the normalized architecture name for arm64.
+	ArchitectureArm64 NormalizedArch = "arm64"
+)
+
+// machineTypePrefixArchitectureMap contains a map of (machineTypePrefix, architecture) tuples
+var machineTypePrefixArchitectureMap = map[string]NormalizedArch{
+	"t2a": ArchitectureArm64,
```

Comment:
better to maintain a curated file with a list of all instance types and fetch the info from there rather than do this?

Reply:
Do you mean a file to unmarshal at need or to add more families to this map?

Reply:
i thought a file with all the families possible from the cmdline. But if that is not possible, we might have to live with this.

Reply:
How frequently are new instance families likely to be added such that we are going to need to update this? I guess there's no API that GCP exposes currently that gives you a sense of the architecture of an instance type or family?

Reply:
No, no exposed API to gather the architecture of a given machine type unfortunately.

As for the frequency, I'm not sure, the first arm64 instances have been released this year.